sw/device/tests/clkmgr_external_clk_src_for_sw_impl.h

```diff
@@ -0,0 +1,21 @@
+// Copyright lowRISC contributors.
```

Comment:
@timothytrippel 
is this okay in your mind? Or do you prefer this as a testutil?
I think it's fine outside of testutil..since this is a WHOLE test that is meant to support different variants.

Reply:
Been discussing with @matutem, I think this is fine too since it's the whole test. I think the organization of these sorts of things is hard to determine a priori. This seems sensible, and if it ever causes an issue we can always reorganize it.